### ISSUE TITLE
Generation of metadata fails when an entity has two properties with the same owned type.

### ISSUE THREAD

**Issue by apircher** [Breeze/breeze.server.net#186]
I have an owned type called ChangeInfo:

```
[Owned]
public class ChangeInfo
{
    [MaxLength(256)]
    public required string By { get; set; }
    public DateTimeOffset At { get; set; }
}
```

And an entity which uses this type for creation and modification info:

```
public class MyEntity
{
    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    ...

    public required ChangeInfo Created { get; set; }

    public required ChangeInfo Modified { get; set; }

    ...
}
```

In Program.cs I try to create metadata for a DbContext which has a DbSet of this entity:

```
if (args.Length > 0 && args[0].Contains("metadata"))
{
    // Generate breeze metadata and exit.
    using (var serviceScope = app.Services.CreateScope())
    {
        var services = serviceScope.ServiceProvider;
        var context = services.GetRequiredService<MyDbContext>();
        var persistanceManager = new MyDbPersistenceManager(context);
        var metadata = persistanceManager.Metadata();
        Console.Out.WriteLine(metadata);
    }
    return;
}
```

But it fails with an ArgumentException which says **"An item with the same key has already been added. Key: ChangeInfo"**.

The problem seems to occur in [Breeze.Persistence.EFCore.MetadataBuilder in line 31](https://github.com/Breeze/breeze.server.net/blob/87e248842bd6db621567245bc2eb4c06dc8faece/AspNetCore/Breeze.Persistence.EFCore/MetadataBuilder.cs#L31).

I can reproduce the same exception with the following code:

```
using (var serviceScope = app.Services.CreateScope())
{
    var services = serviceScope.ServiceProvider;
    var context = services.GetRequiredService<MyDbContext>();

    var complexTypes = context.Model.GetEntityTypes()
        .Where(et => et.IsOwned())
        .Select(et => et.ClrType.Name).ToList();
    var map = complexTypes.ToDictionary(mt => mt);
}
```

This is quite similar to what happens in the [MetadataBuilder from line 26 to 28.](https://github.com/Breeze/breeze.server.net/blob/87e248842bd6db621567245bc2eb4c06dc8faece/AspNetCore/Breeze.Persistence.EFCore/MetadataBuilder.cs#L26) In this case the complexTypes list will contain two MetaType instances with a ShortName of 'ChangeInfo' and so the call ``complexTypes.ToDictionary(mt => mt.ShortName)`` throws the exception mentioned above.

I am using:
- Breeze.AspNetCore.NetCore 7.0.2
- Breeze.Persistence.EFCore 7.0.2
- Microsoft.EntityFrameworkCore.Tools 7.0.4
- Microsoft.EntityFrameworkCore.SqlServer 7.0.4

Any help would be greatly appreciated.

**Comment by steveschmitt** [comment]
Yes, it looks like a bug.  We will get a fix in soon.  Thanks for finding this and letting us know.

**Comment by steveschmitt** [comment]
Fixed in version 7.1.0
